src/Pool.sol

```diff
@@ -95,15 +112,37 @@ contract Pool is Ownable {
     //
     function setAddresses(address _chickenBondManagerAddress) external onlyOwner {
         chickenBondManagerAddress = _chickenBondManagerAddress;
-        renounceOwnership();
     }
 
     fallback() external payable {}
     receive() external payable {}
 
-    function updatePool(address _storageProvider, uint256 amount) public {
-        lockedCapital[_storageProvider] += amount / 2;
-        totalWorkingCapital += amount / 2;
+    function harvestRewards() public {
+        updatePool(address(0), 0);
+
+        PoolStaker storage staker = poolStakers[msg.sender];
+        uint256 rewards = staker.amount * accumulatedRewardsPerShare / REWARDS_PRECISION;
+        uint256 rewardsToHarvest = rewards - staker.rewardDebt;
+
+        staker.rewardDebt = rewards;
+
+        if (rewardsToHarvest == 0) {
+            return;
+        }
+
+        emit HarvestRewards(msg.sender, rewardsToHarvest);
+        // payable(manager).transfer(rewardsToHarvest);
+    }
+
+    function updatePool(address _storageProvider, uint256 rewards) public {
```

Comment:
Why does this need to be a `public` function ?

Reply:
Maybe it doesn't need to be public.
The broker needs to call it but we can have more fine grained control later on

Reply:
If its only the broker it should be `external`, also if it doesn't have any authorization it means that anyone can call this function and update storageProviders with random rewards so some kind of broker mapping and require would be a good idea

Reply:
I'll change this to `external` now but will only take care of proper authorization after everything is set in place